File: test_data/csharp/test_main/aas_core_meta.v3/input/snippets/Verification/properties_or_ranges_have_value_type.cs
/// <summary>
/// Check that the <paramref name="elements" /> which are
/// <see cref="Aas.IProperty" /> or <see cref="Aas.IRange" />
/// have the given <paramref name="valueType" />.
/// </summary>
/// <remarks>
/// We have to use nullable valueType since the compiler does not really handle
/// nullable C# value types.
///
/// See https://endjin.com/blog/2022/02/csharp-10-generics-nullable-references-improvements-allownull
/// </remarks>
public static bool PropertiesOrRangesHaveValueType(
    IEnumerable<Aas.ISubmodelElement> elements,
    Aas.DataTypeDefXsd? valueType
)
{
    foreach (var element in elements)
    {
        switch (element)
        {
            case Aas.IProperty prop:
                if (prop.ValueType != valueType)
                {
                    return false;
                }
                break;
            case Aas.IRange range:
                if (range.ValueType != valueType)
                {
                    return false;
                }
                break;
        }
    }
    return true;
}

File: test_data/java/test_main/aas_core_meta.v3/input/snippets/Verification/properties_or_ranges_have_value_type.java
/**
* Check that the elements which are
* {@link IProperty} or {@link IRange}
* have the given valueType.
 * @param elements the elements.
 * @param valueType the valueType.
*/
public static boolean propertiesOrRangesHaveValueType(
  Iterable<? extends ISubmodelElement> elements,
  DataTypeDefXsd valueType
) {
  Objects.requireNonNull(elements);
  Objects.requireNonNull(valueType);

  for (ISubmodelElement element : elements) {
     if(element instanceof IProperty) {
       if (((IProperty) element).getValueType() != valueType) {
         return false;
       }
     } else if (element instanceof IRange) {
       if (((IRange) element).getValueType() != valueType) {
         return false;
       }
     }
  }
  return true;
}
